crates/sui/src/sui_move/new.rs

```diff
@@ -1,16 +1,25 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use move_cli::package::cli::create_move_package;
-use std::path::Path;
+use clap::Parser;
+use move_cli::base::new;
+use std::path::PathBuf;
 
-pub fn execute(path: &Path, name: &String) -> anyhow::Result<()> {
-    create_move_package(path,
-                        name,
+#[derive(Parser)]
+pub struct New {
+    #[clap(flatten)]
+    pub new: new::New,
+}
+
+impl New {
+    pub fn execute(self, path: Option<PathBuf>) -> anyhow::Result<()> {
+        let name = &self.new.name.to_lowercase();
```

Comment:
Nit, unpack the command so newly added fields aren't ignored

Reply:
If I unpack than I cannot call `self.new` anymore due to partially moved value and `new::New` on the core Move side is not clone-able - I am not sure how to work around it...

Reply:
You can do `let New { name } = &self`

Reply:
It still does not work as core Move side's `New::execute` takes `self` by values so we cannot borrow it in the unpack and then use it as the argument...